src/foam/nanos/notification/email/DefaultPropertyEmailTemplateService.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.notification.email',
+  name: 'DefaultPropertyEmailTemplateService',
+
+  documentation: 'Used in conjuction with ChainedPropertyService',
+
+  implements: [
+    'foam.nanos.notification.email.EmailPropertyService'
+  ],
+
+  javaImports: [
+    'foam.nanos.app.EmailConfig',
+    'foam.nanos.logger.Logger',
+    'foam.util.SafetyUtil'
+  ],
+
+  methods: [
+    {
+      name: 'apply',
+      type: 'foam.nanos.notification.email.EmailMessage',
+      args: [
+        {
+          name: 'x',
+          type: 'Context',
+        },
+        {
+          name: 'group',
+          class: 'String',
+          documentation: 'group of user whose the recipient of the email being sent'
+        },
+        {
+          name: 'emailMessage',
+          type: 'foam.nanos.notification.email.EmailMessage',
+          documentation: 'Email message'
+        },
+        {
+          name: 'templateArgs',
+          type: 'Map',
+          documentation: 'Template arguments'
+        }
+      ],
+      javaCode: `
+        Logger logger = (Logger) x.get("logger");
+        EmailConfig emailConfig = (EmailConfig) x.get("emailConfig");
+
+        // Service property check
+        if ( emailConfig == null
+          || SafetyUtil.isEmpty(emailConfig.getReplyTo())
```

Comment:
I don't think making all these checks is necessary and it also doesn't scale well if/when more properties are added to the email config. Also, an empty string for display name is technically valid. I'm not sure replyTo is even a required field for emails either.